pkg/tempo/automaticloggingprocessor/automaticloggingprocessor.go

```diff
@@ -85,12 +87,18 @@ func newTraceProcessor(nextConsumer consumer.Traces, cfg *AutomaticLoggingConfig
 	cfg.Overrides.DurationKey = override(cfg.Overrides.DurationKey, defaultDurationKey)
 	cfg.Overrides.TraceIDKey = override(cfg.Overrides.TraceIDKey, defaultTraceIDKey)
 
+	labels := make(map[string]bool, len(cfg.Labels))
```

Comment:
`map[string]struct{}` is i think the "right" way to do a flag map like this

Reply:
`map[string]struct{}` uses less memory since an empty struct is 0 bytes, but it's very similar in performance to `map[string]bool` (see https://gist.github.com/davecheney/3be245c92b61e5045f75). I think `map[string]bool` makes for a bit clearer code.

Reply:
FWIW I try to use `map[string]struct{}` whenever I can, because there's always a chance that a bug could creep in that sets an entry to `false` in the map which would be treated as if it didn't exist.

Reply:
That's a good point, changed in 7efae7e3517625a38c1d3c769b5763f53091f770